subsys/bluetooth/mesh/proxy.c

```diff
@@ -899,73 +906,67 @@ bool bt_mesh_proxy_relay(struct net_buf_simple *buf, uint16_t dst)
 			continue;
 		}
 
+		if (client->filter_type == PROV) {
+			BT_ERR("Invalid PDU type for Proxy Client");
+			return -EINVAL;
+		}
+
 		/* Proxy PDU sending modifies the original buffer,
 		 * so we need to make a copy.
 		 */
 		net_buf_simple_reserve(&msg, 1);
 		net_buf_simple_add_mem(&msg, buf->data, buf->len);
 
-		bt_mesh_proxy_send(client->conn, BT_MESH_PROXY_NET_PDU, &msg);
+		proxy_segment_and_send(client->conn, BT_MESH_PROXY_NET_PDU,
+					&msg, buf_send_end, net_buf_ref(buf));
```

Comment:
I believe we'll have a memory leak here now if the `proxy_send` calls fail to send, for instance by returning early with the `if (!attr)` check on line 952. I think that check needs to return an error code, and this line needs to check the return code of this function, and call `net_buf_unref` inline if it fails.

Reply:
It was modified from the beginning https://github.com/zephyrproject-rtos/zephyr/pull/32666#discussion_r584540793